### DISCUSSION TITLE
Thanos Querier - Some requests are unathorized 401

### DISCUSSION THREAD

**Discussion by alinjie** [grafana/grafana#54693] [Data Sources]
Hi

I'm trying to set up a Prometheus datasource to fetch metrics from OpenShift's built-in Thanos querier. I've added a custom `Authorization` header to the datasource and when saving/testing the datasource configuration, everyhing seems fine. When I go to the `Explore` page and select the newly created datasource, I get the following error in Grafana.

![image](https://user-images.githubusercontent.com/38862494/188402347-090ec6ee-a8fc-4f28-b93e-8d9aa4cff0c5.png)

Querying the datasource works fine and I get the expected results back, but Grafana is however unable to list the metric names, which I suspect is the reason for the error above. When checking the network tab in my browser I can see that the `/api/v1/labels` request returns a `401`.

When running the exact same request (`GET /api/v1/labels`) directly to the Thanos querier, with the same `Authorization`header, I get the expected results back.

Is Grafana excluding the configured custom HTTP headers when running the metric lookup requests or have I misconfigured the datasource?

**Comment by alinjie** [comment]
FYI: I'm running Grafana 9.0.0

**Reply by okyspace** [reply]
I have same issue but haven't found solution online yet. Did u manage to solve?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
